SocialCareCaseViewerApi/V1/Gateways/DatabaseGateway.cs

```diff
@@ -1032,6 +1045,45 @@ public Infrastructure.CaseStatusType GetCaseStatusTypeWithFields(string type)
             return response.FirstOrDefault();
         }
 
+        public Infrastructure.CaseStatus CreateCaseStatus(CreateCaseStatusRequest request)
+        {
+            var caseStatus = new Infrastructure.CaseStatus()
+            {
+                PersonId = request.PersonId,
+                TypeId = (long) request.TypeId,
+                StartDate = request.StartDate,
+                EndDate = request.EndDate,
+                Notes = request.Notes,
+                CreatedBy = request.CreatedBy
+            };
+
+            _databaseContext.CaseStatuses.Add(caseStatus);
+            _databaseContext.SaveChanges();
+
+            foreach (var optionValue in request.Fields)
+            {
+                var field = _databaseContext.CaseStatusTypeFields
+                .Where(f => f.Name == optionValue.Name)
+                .FirstOrDefault();
+
+                var fieldTypeOption = _databaseContext.CaseStatusTypeFieldOptions
+                .Where(fov => fov.Name == optionValue.Selected)
+                .Where(fov => fov.TypeFieldId == field.Id)
+                .FirstOrDefault();
+
+                CaseStatusFieldOption fieldOptions = new Infrastructure.CaseStatusFieldOption()
+                {
+                    StatusId = caseStatus.Id,
+                    FieldOptionId = fieldTypeOption.Id
```

Comment:
fieldTypeOptions could be null, .FirstOrDefault returns null if nothing is found, how do we want to handle this?

Trying to access .Id of null will end up with a 500 response currently.

Reply:
Added null check